docs/reference/modules/cluster/shards_allocation.asciidoc

```diff
@@ -49,9 +49,21 @@ one of the active allocation ids in the cluster state.
 `cluster.routing.allocation.same_shard.host`::
       (<<dynamic-cluster-setting,Dynamic>>)
       Allows to perform a check to prevent allocation of multiple instances of
-      the same shard on a single host, based on host name and host address.
-      Defaults to `false`, meaning that no check is performed by default. This
-      setting only applies if multiple nodes are started on the same machine.
+      the same shard on a single host, based on host name and host address. If
+      the host address is not empty, the addresses of the two nodes are used
+      to determine whether they are on the same host. Otherwise, use the host
+      name to distinguish. Defaults to `false`, meaning that no check is performed
+      by default. This setting only applies if multiple nodes are started on
+      the same machine.
+
+[[cluster-routing-allocation-same-shard-host-name]]
+`cluster.routing.allocation.same_shard.host_name`::
+      (<<dynamic-cluster-setting,Dynamic>>)
+      Only works when `cluster.routing.allocation.same_shard.host` is `true`.
```

Comment:
This wasn't what I intended to mean in https://github.com/elastic/elasticsearch/pull/80702#discussion_r750090527. I think users should generally set _one of_ `cluster.routing.allocation.same_shard.host` or `cluster.routing.allocation.same_shard.host_name`, depending on whether they want to separate shards by address or by name.

Reply:
Can these two settings be true at the same time? In this way, users can distinguish hosts by address and name.

Reply:
That's up to you as the implementor. I'm not really sure where that would be useful.